homeassistant/components/flume/sensor.py

```diff
@@ -137,26 +148,49 @@ def available(self):
 
     @property
     def unique_id(self):
-        """Device unique ID."""
-        return self._device_id
+        """Flume query and Device unique ID."""
+        return f"{self._flume_query_sensor}_{self._device_id}"
 
-    @Throttle(MIN_TIME_BETWEEN_UPDATES)
     def update(self):
         """Get the latest data and updates the states."""
         _LOGGER.debug("Updating flume sensor: %s", self._name)
-        try:
-            self._flume_device.update_force()
-        except Exception as ex:  # pylint: disable=broad-except
-            if self._available:
-                _LOGGER.error("Update of flume sensor %s failed: %s", self._name, ex)
-            self._available = False
-            return
+
+        def format_attribute_value(value):
+            return round(value, 1) if isinstance(value, Number) else None
+
+        self._flume_data.update()
+        self._state = format_attribute_value(
+            self._flume_data.flume_device.values[self._flume_query_sensor]
+        )
         _LOGGER.debug("Successful update of flume sensor: %s", self._name)
-        self._state = self._flume_device.value
+
         self._available = True
```

Comment:
Should we use `self._flume_data.available`?

Reply:
Updated